### PR TITLE
chore: upgrade webpack to 4.5.0

### PR DESCRIPTION
**Pull request by efloden**
JIRA issue URL: N/A

Upgrade webpack to version 4.50
Upgrade all affected webpack plugins and packages
Extract storybook webpack config to separate file to maintain backwards compatibility

## QA 
Some smoke testing of all builds is suggested:
rundev
make watch
make watch-editor
make storybook-editor
make storybook-frontend

## Checklist

- JIRA link
- Check target branch
- Make sure all commit statuses are green or otherwise documented reasons to ignore
- QA needs to evaluate against the JIRA ticket
- Changed files and commits make sense for this PR

See [Zanata Development Guidelines](https://github.com/zanata/zanata-platform/wiki/Development-Guidelines) more for information.

----
*This template can be updated in .github/PULL_REQUEST_TEMPLATE.md*

### CODE REVIEW

File: server/zanata-frontend/src/webpack.config.js [hunk 1]
```diff
@@ -258,15 +261,6 @@ module.exports = function (env, isEditor, devServerPort) {
         // storybook should use the fallback: style-loader
         disable: storybook || dev
       }),
-      new webpack.NoEmitOnErrorsPlugin(),
-
-      prod
-        ? new webpack.optimize.UglifyJsPlugin({ sourceMap: true })
```
Comment (seanf) [inline server/zanata-frontend/src/webpack.config.js]: I don't see anything about minimisation now. Is this just implicit for 'production' now?

Comment (efloden) [reply server/zanata-frontend/src/webpack.config.js]: Yes, minimization and a host of other options appear to be enabled by default now , which explains how they can claim such big performance gains with webpack 4: https://medium.com/webpack/webpack-4-mode-and-optimization-5423a6bc597a

**Comment by seanf** [comment]
:lgtm:

---

Reviewed 3 of 3 files at r2.
Review status: all files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://beta.reviewable.io/reviews/zanata/zanata-platform/776#-:-LARbeq7Aty2hirHMWn1:bnfp4nl)*

**Comment by seanf** [comment]
@efloden Probably best if we have a smidge of smoke-test QA on this one.

**Comment by seanf** [comment]
Reviewed 6 of 7 files at r3.
Review status: 6 of 7 files reviewed at latest revision, all discussions resolved, some commit checks failed.

---

*[server/zanata-frontend/src/webpack.config.storybook.js, line 3 at r3](https://reviewable.io/reviews/zanata/zanata-platform/776#-LApbuld4KG0b3k1pzSP:-LApbuld4KG0b3k1pzSQ:b-xlsa5h) ([raw file](https://github.com/zanata/zanata-platform/blob/47333e0983f69818f6b3045c36a881c2f5764fa5/server/zanata-frontend/src/webpack.config.storybook.js#L3)):*
> ```JavaScript
> /**
>  * This is the base config for Storybook builds. It does not utilize the latest
>  * webpack features for backwards compatibility with storybook/react.
> ```

This looks like a lot of duplicated config to keep in sync. Is there some way of extracting some of the commonalities?

Or perhaps it would be worth trying storybook v4.0.0-alpha.3 - https://github.com/storybooks/storybook/issues/3044

Yes, it's an alpha, but at least it's not a runtime dependency we're talking about.

---


*Comments from [Reviewable](https://reviewable.io/reviews/zanata/zanata-platform/776)*

**Comment by efloden** [comment]
Review status: 6 of 7 files reviewed at latest revision, 1 unresolved discussion, some commit checks failed.

---

*[server/zanata-frontend/src/webpack.config.storybook.js, line 3 at r3](https://reviewable.io/reviews/zanata/zanata-platform/776#-LApbuld4KG0b3k1pzSP:-LAq3e_nAZU6fjOyeXrC:blvlonq) ([raw file](https://github.com/zanata/zanata-platform/blob/47333e0983f69818f6b3045c36a881c2f5764fa5/server/zanata-frontend/src/webpack.config.storybook.js#L3)):*
<details><summary><i>Previously, seanf (Sean Flanigan) wrote…</i></summary><blockquote>

This looks like a lot of duplicated config to keep in sync. Is there some way of extracting some of the commonalities?

Or perhaps it would be worth trying storybook v4.0.0-alpha.3 - https://github.com/storybooks/storybook/issues/3044

Yes, it's an alpha, but at least it's not a runtime dependency we're talking about.
</blockquote></details>

Trying the storybook v4.0.0-alpha.3, and the storybook-editor appears to run without any problems - but storybook-frontend has this error: 
```
react_2.action is not a function
TypeError: react_2.action is not a function
    at Object.eval (webpack-internal:///./app/components/RejectionsForm/RejectionsForm.story.js:15:23)
    at eval (webpack-internal:///./app/components/RejectionsForm/RejectionsForm.story.js:28:30)
    at Object../app/components/RejectionsForm/RejectionsForm.story.js (http://localhost:9001/static/preview.bundle.js:1255:1)
    at __webpack_require__ (http://localhost:9001/static/preview.bundle.js:712:30)
    at fn (http://localhost:9001/static/preview.bundle.js:95:20)
    at eval (webpack-internal:///./app/components/components.story.js:32:1)
    at Object../app/components/components.story.js (http://localhost:9001/static/preview.bundle.js:1446:1)
    at __webpack_require__ (http://localhost:9001/static/preview.bundle.js:712:30)
    at fn (http://localhost:9001/static/preview.bundle.js:95:20)
    at loadStories (webpack-internal:///./.storybook-frontend/config.js:67:3)
```
Tried updating the storybook types, not sure what the problem is in RejectionsForm.story.js

---


*Comments from [Reviewable](https://reviewable.io/reviews/zanata/zanata-platform/776)*

**Comment by efloden** [comment]
Review status: 3 of 7 files reviewed at latest revision, 1 unresolved discussion, some commit checks broke.

---

*[server/zanata-frontend/src/webpack.config.storybook.js, line 3 at r3](https://reviewable.io/reviews/zanata/zanata-platform/776#-LApbuld4KG0b3k1pzSP:-LAq7l3-CPe8qgTtbktE:b-68d5qv) ([raw file](https://github.com/zanata/zanata-platform/blob/47333e0983f69818f6b3045c36a881c2f5764fa5/server/zanata-frontend/src/webpack.config.storybook.js#L3)):*
<details><summary><i>Previously, efloden (Earl Floden) wrote…</i></summary><blockquote>

Trying the storybook v4.0.0-alpha.3, and the storybook-editor appears to run without any problems - but storybook-frontend has this error: 
```
react_2.action is not a function
TypeError: react_2.action is not a function
    at Object.eval (webpack-internal:///./app/components/RejectionsForm/RejectionsForm.story.js:15:23)
    at eval (webpack-internal:///./app/components/RejectionsForm/RejectionsForm.story.js:28:30)
    at Object../app/components/RejectionsForm/RejectionsForm.story.js (http://localhost:9001/static/preview.bundle.js:1255:1)
    at __webpack_require__ (http://localhost:9001/static/preview.bundle.js:712:30)
    at fn (http://localhost:9001/static/preview.bundle.js:95:20)
    at eval (webpack-internal:///./app/components/components.story.js:32:1)
    at Object../app/components/components.story.js (http://localhost:9001/static/preview.bundle.js:1446:1)
    at __webpack_require__ (http://localhost:9001/static/preview.bundle.js:712:30)
    at fn (http://localhost:9001/static/preview.bundle.js:95:20)
    at loadStories (webpack-internal:///./.storybook-frontend/config.js:67:3)
```
Tried updating the storybook types, not sure what the problem is in RejectionsForm.story.js
</blockquote></details>

This error was due to the file importing action from @storybook/react rather than @storybook/addon-actions

---


*Comments from [Reviewable](https://reviewable.io/reviews/zanata/zanata-platform/776)*

**Comment by efloden** [comment]
Review status: 3 of 7 files reviewed at latest revision, 1 unresolved discussion, some commit checks broke.

---

*[server/zanata-frontend/src/webpack.config.storybook.js, line 3 at r3](https://reviewable.io/reviews/zanata/zanata-platform/776#-LApbuld4KG0b3k1pzSP:-LAqH3Rn5DRG7VaCW1Bb:b-mwwrz1) ([raw file](https://github.com/zanata/zanata-platform/blob/47333e0983f69818f6b3045c36a881c2f5764fa5/server/zanata-frontend/src/webpack.config.storybook.js#L3)):*
<details><summary><i>Previously, efloden (Earl Floden) wrote…</i></summary><blockquote>

This error was due to the file importing action from @storybook/react rather than @storybook/addon-actions
</blockquote></details>

storybook/react rather than storybook/addon-actions *

---


*Comments from [Reviewable](https://reviewable.io/reviews/zanata/zanata-platform/776)*

**Pull request closed without merging**